code/parse/sexp.cpp

```diff
@@ -1968,6 +1963,33 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, i
 			continue; 
 		}
 
+		// if this is the special argument, make sure it is being properly used
+		if (type2 == SEXP_ATOM_STRING && Sexp_nodes[node].flags & SNF_SPECIAL_ARG_IN_NODE) {
+			bool found = false;
+			z = node;
+			while (true) {
+				z = find_parent_operator(z);
+				if (z < 0) {
```

Comment:
I take it that `z < 0` means we've gone all the way up the SEXP tree without finding what we're looking for?

Reply:
yup